File: 2D-World-Simulator-Python/OPP-Python/Organisms/Plants/Guarana.py
from Game.OrganismStatics import *
from Game.Plant import Plant


class Guarana(Plant):
    def __init__(self, organism_world, point):
        super().__init__(organism_world, point, 0, 0, "+", OrganismSpecie.GUARANA, "#ff2b1c")

    # checking defence successfulness
    def defence(self, attacking_organism):
        self.increase_strength(attacking_organism)
        # plant must be eaten anyway -> means we come back to collision()
        return False

    # performing defence of guarana -> adding strength to attacking organism
    def increase_strength(self, attacking_organism):
        new_strength = attacking_organism.get_strength() + ADDITIONAL_STRENGTH
        attacking_organism.set_strength(new_strength)
        self.organism_world.get_world_window().add_comment(attacking_organism.get_organism_sign() +
                                                           " gained +3 STRENGTH by eating GUARANA!")

File: 2D-World-Simulator-Java/OPP-Java/src/op/JavaProject/Organisms/Plants/Guarana.java
package op.JavaProject.Organisms.Plants;

import op.JavaProject.Functionality.BoardPoint;
import op.JavaProject.Functionality.Species;
import op.JavaProject.Game.Organism;
import op.JavaProject.Game.Plant;
import op.JavaProject.Game.World;

import java.awt.*;

import static op.JavaProject.Game.OrganismsStatics.ADDITIONAL_STRENGTH;

public class Guarana extends Plant {

    public Guarana(World organismWorld, BoardPoint point) {
        super(organismWorld, point);
        this.strength = 0;
        this.initiative = 0;
        this.organismSign = "+";
        this.organismSpecie = Species.GUARANA;
        this.organismColor = new Color(255, 43, 28);
    }

    //checking defence successfulness
    @Override
    public boolean Defence(Organism attackingOrganism) {

        if (IncreaseStrength(attackingOrganism)) return true;
        return false;
    }

    //performing defence of guarana -> adding strength to attacking organism
    private boolean IncreaseStrength(Organism attackingOrganism) {
        int currentStrength = attackingOrganism.GetStrength();
        attackingOrganism.SetStrength(currentStrength + ADDITIONAL_STRENGTH);
        organismWorld.GetCommentator().AddComment(attackingOrganism.GetOrganismSign() + " gained +3 STRENGTH by eating GUARANA!");
        //plant must be eaten anyway -> means we need to come back to collision()
        return false;
    }

}



